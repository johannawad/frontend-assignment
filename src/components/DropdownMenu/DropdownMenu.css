.menu{
    margin: 0 20px 20px 20px;
    list-style-type: none;
    border: 1px solid #ccc;
    
}

.menu__item{
    padding: 10px 10px;
    color: grey;
    border-top: 1px solid #ccc;
}

li:first-of-type {
    border-top: none;
}

.menu__item__content {
    display: flex;
}