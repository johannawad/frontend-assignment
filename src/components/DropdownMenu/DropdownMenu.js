import React from 'react';
import './DropdownMenu.css';
import {highlightQueryMatch} from '../../utils/textFormatter';

const DropdownMenu = ({items, query}) => {
    return (
        <div className="menu">
            {items?.map((item, index) => 
                <li className="menu__item" key={index}>
                    <div className="menu__item__content">
                        {item.title? highlightQueryMatch(item.title, query): null}
                        {item.quantity? ` (${item.quantity})`: null}
                    </div>
                </li>
            )}
        </div>
    )
}

export default DropdownMenu;