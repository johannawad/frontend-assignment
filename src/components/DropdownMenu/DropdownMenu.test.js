import React from 'react';
import DropdownMenu from './DropdownMenu';
import renderer from 'react-test-renderer';

describe('SearchBar', () => {
    it('renders correctly without any items', () => {
        const tree = renderer.create(<DropdownMenu />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders correctly with items', () => {
        const items = [{title: 'Title', quantity: 1}, {title: 'Title', quantity: 1}]
        const tree = renderer.create(<DropdownMenu items={items} />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders without children when items do not have title and quantity props', () => {
        const items = [{someProp: 'Title', someOtherProp: 1}]
        const tree = renderer.create(<DropdownMenu items={items} />).toJSON();
        const content = tree.children[0].children[0].children;
        expect(content).toBeNull();
    })


    it('renders with 2 children when items have title and quantity props', () => {
        const items = [{title: 'Title', quantity: 1}]
        const tree = renderer.create(<DropdownMenu items={items} />).toJSON();
        const content = tree.children[0].children[0].children;
        expect(content.length).toEqual(2);
    })
})