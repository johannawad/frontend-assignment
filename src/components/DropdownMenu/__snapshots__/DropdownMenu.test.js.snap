// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly with items 1`] = `
<div
  className="menu"
>
  <li
    className="menu__item"
  >
    <div
      className="menu__item__content"
    >
      <div>
        <b>
          title
        </b>
        
      </div>
       (1)
    </div>
  </li>
  <li
    className="menu__item"
  >
    <div
      className="menu__item__content"
    >
      <div>
        <b>
          title
        </b>
        
      </div>
       (1)
    </div>
  </li>
</div>
`;

exports[`SearchBar renders correctly without any items 1`] = `
<div
  className="menu"
/>
`;
